spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryProperties.java

```diff
@@ -70,7 +79,8 @@ private ZookeeperDiscoveryProperties() {}
 
 	public ZookeeperDiscoveryProperties(InetUtils inetUtils) {
 		this.hostInfo = inetUtils.findFirstNonLoopbackHostInfo();
-		this.instanceHost = this.hostInfo.getIpAddress();
+		this.instanceHost = this.hostInfo.getHostname();
+		this.instanceIpAddress = this.hostInfo.getIpAddress();
```

Comment:
@marcingrzejszczak  I'm not sure when will not set instanceIpAddress. If not set instanceIpAddress , it will get the address from hostInfo.